src/components/KInput/KInput.vue

```diff
@@ -237,21 +195,21 @@ const helpText = computed((): string => {
   }
 
   // if error prop is true and there is an error message, return that
-  if ((props.error || props.hasError) && props.errorMessage) {
-    return props.errorMessage
+  if ((error || hasError) && errorMessage) {
+    return errorMessage
   }
 
   // otherwise return the help text
   // if error prop is true it danger styles will be applied
-  return props.help
+  return help
 })
 
 watch(charLimitExceeded, (newVal, oldVal) => {
   if (newVal !== oldVal) {
     emit('char-limit-exceeded', {
       value: currValue.value,
       length: currValue.value.length,
-      characterLimit: props.characterLimit,
+      characterLimit: characterLimit!,
```

Comment:
Could this be `undefined` or `null`?

Reply:
@Justineo could be, you can see in the screenshot, if the value `characterLimit` is not a valid value, this watcher will not be triggered.
![CleanShot 2025-04-17 at 14 36 13@2x](https://github.com/user-attachments/assets/31fcad1b-536f-4bfb-a291-a5ae7870e686)